cmd/etcd-launcher/main.go

```diff
@@ -88,30 +89,40 @@ func main() {
 		for { // reconcile dead members
 			members, err := e.listMembers()
 			if err != nil {
+				log.Warnf("failed to list memebers: %v ", err)
 				time.Sleep(10 * time.Second)
 				continue
 			}
 			// we only need to reconcile if we have more members than we should
 			if len(members) <= e.config.clusterSize {
+				log.Info("cluster members reconciled..")
 				break
 			}
 			// to avoide race conditions, we will run only on the cluster leader
 			leader, err := e.isLeader()
 			if err != nil || !leader {
+				if err != nil {
```

Comment:
You don't need to check the error two times, instead make the log message more general: 
`log.Warnw("failed to remove member, error occurred or didn't get the current leader", zap.Error(err))`

In this case, if there was an error it will be printed out as an error